README.md

```diff
@@ -1,207 +1,607 @@
+| `panoptic-segmentation` | *n/a*                       | Specific segmentation task that combines instance segmentation of objects and semantic labels for non-objects.  |
+| `similarity-search`     | *n/a*                       | Generic task to identify whether a query input corresponds to another reference within a corpus.                |
+| `generative`            | *n/a*                       | Generic task that encompasses all synthetic data generation techniques.                                         |
+| `image-captioning`      | *n/a*                       | Specific task of describing the content of an image in words.                                                   |
+| `super-resolution`      | *n/a*                       | Specific task that increases the quality and resolution of an image by increasing its high-frequency details.   |
+
+If the task falls within the category of supervised machine learning and uses labels during training,
+this should align with the `label:tasks` values defined in [STAC Label Extension][stac-ext-label-props] for relevant
+STAC Collections and Items published with the model described by this extension.
+
+It is to be noted that multiple "*generic*" tasks names (`classification`, `detection`, etc.) are defined to allow
+correspondance with `label:tasks`, but these can lead to some ambiguity depending on context. For example, a model
+that supports `classification` could mean that the model can predict patch-based classes over an entire scene
+(i.e.: `scene-classification` for a single prediction over an entire area of interest as a whole),
+or that it can predict pixel-wise "classifications", such as land-cover labels for
+every single pixel coordinate over the area of interest. Maybe counter-intuitively to some users,
+such a model that produces pixel-wise "classifications" should be attributed the `segmentation` task
+(and more specifically `semantic-segmentation`) rather than `classification`. To avoid this kind of ambiguity,
+it is strongly recommended that `tasks` always aim to provide the most specific definitions possible to explicitly
+describe what the model accomplishes.
 
-#### Process Level Enum
-
-It is recommended to use the [STAC Processing Extension][stac-ext-proc]
-to represent the `processing:level` of the relevant level `L0` for raw data up to `L4` for Analysis-Ready Data (ARD).
+[stac-ext-label-props]: https://github.com/stac-extensions/label#item-properties
 
-[stac-ext-proc]: https://github.com/stac-extensions/processing#suggested-processing-levels
+### Framework
+
+In most cases, this should correspond to common library names of well-established ML frameworks.
+No explicit "Enum" is defined to allow easy addition of newer frameworks, but it is recommended
+to use common names when applicable. Below are a few notable entries.
+
+- `PyTorch`
+- `TensorFlow`
+- `Scikit-learn`
+- `Huggingface`
+- `PyMC`
+- `JAX`
+- `ONNX`
+- `MXNet`
+- `Keras`
+- `Caffe`
+- `Weka`
```

Comment:
should `Weka` be listed? I've never heard of it or seen it in the wild.

Here's a suggested reordering based on my subjective interpretation of current popularity + longevity. 

I also added `rgee` and `spatialRF` to showcase some R options. especially in academia, lots of folks use R, particularly random forest models for semantic segmentation.

I removed Caffe (no updates in 4 years) and MxNet (archived last year). I don't think anyone will publish models for these frameworks.

Removed ONNX since it isn't a training framework and I think the purpose of this field is to describe the framework used to train the model. this might be different than the inference runtime and format.

```suggestion
- `PyTorch`
- `TensorFlow`
- `Scikit-learn`
- `Huggingface`
- `Keras`
- `rgee`
- `spatialRF`
- `JAX`
- `PyMC`
```

Reply:
Weka is not as common as others, but I have seen it once or twice. I prefer to have the option than not. 
I think it is a good idea to include ONNX. Remember that MLM is not for training only. It can be used for inference runtime. 
I don't think popularity or older frameworks should be ignored. People might want to support older algorithms that work well. 
Will add the R-based references, those are good examples.

Reply:
Sure I'm fine to include Weka then and include the old frameworks.

> I think it is a good idea to include ONNX. Remember that MLM is not for training only. It can be used for inference runtime.

I totally agree (in fact I don't think MLM is a good fit for describing how to train a model), but we call out that this `framework` field should denote the framework used for training in the table. ONNX isn't used to train ML models. Instead, that can be denoted in the asset details. I f we include it, someone could get confused and not supply the actual framework used for training.